File: nhglib-core/src/io/github/movementspeed/nhglib/input/models/impls/system/NhgTouchInput.java
package io.github.movementspeed.nhglib.input.models.impls.system;

import io.github.movementspeed.nhglib.input.models.InputType;
import io.github.movementspeed.nhglib.input.models.base.NhgInput;

public class NhgTouchInput extends NhgInput {
    private int pointerNumber;

    public NhgTouchInput(String name) {
        super(name);
        setType(InputType.TOUCH);
    }

    public int getPointerNumber() {
        return pointerNumber;
    }

    public void setPointerNumber(int pointerNumber) {
        this.pointerNumber = pointerNumber;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/input/models/impls/system/NhgTouchInput.kt
package io.github.movementspeed.nhglib.input.models.impls.system

import com.badlogic.gdx.utils.Array
import io.github.movementspeed.nhglib.input.enums.InputType
import io.github.movementspeed.nhglib.input.enums.TouchInputType
import io.github.movementspeed.nhglib.input.models.base.NhgInput

class NhgTouchInput(name: String) : NhgInput(name) {
    var pointerNumber: Int = 0
    private var touchInputTypes: Array<TouchInputType>? = null

    init {
        type = InputType.TOUCH
    }

    fun hasTouchInputType(touchInputType: TouchInputType): Boolean {
        var res = false

        for (type in touchInputTypes!!) {
            if (type == touchInputType) {
                res = true
            }
        }

        return res
    }

    fun setTouchInputTypes(touchInputTypes: Array<TouchInputType>) {
        this.touchInputTypes = touchInputTypes
    }
}
